src/mds/FSMap.cc

```diff
@@ -224,6 +224,51 @@ void FSMap::print(ostream& out) const
   }
 }
 
+void FSMap::print_daemon_summary(ostream& out) const
+{
+  // this appears in the "services:" section of "ceph status"
+  int num_up = 0, num_in = 0, num_failed = 0;
+  for (auto& [fscid, fs] : filesystems) {
+    num_up += fs->mds_map.get_num_up_mds();
+    num_in += fs->mds_map.get_num_in_mds();
+    num_failed += fs->mds_map.get_num_failed_mds();
+  }
+  int num_standby = standby_daemons.size();
+  out << num_up << "/" << num_in << " daemons up";
+  if (num_failed) {
+    out << " (" << num_failed << " failed)";
+  }
+  if (num_standby) {
+    out << ", " << num_standby << " standby";
+  }
```

Comment:
@batrick Do you think this should provide more detail, like counts of actual states (up:replay vs up:rejoin etc)?

Reply:
I don't think that's useful for most people. Maybe just say the FS is "recovering" if (a) there are no "failed" ranks and (b) not all ranks are active. If they want more info, there is `fs dump`.